docs/shared/recipes/running-tasks/customizing-inputs.md

```diff
@@ -44,136 +48,282 @@ To understand some of the examples below, let's imagine the following simple wor
+
+```jsonc
+"targets": {
+  "lint": {
+    "command": "eslint ."
   }
 }
 ```
 
-{% /tab %}
-{% tab label="Project Level (project.json)" %}
+This default behavior can be overridden by adding any external dependency inputs and will enable Nx to use its cache to be used more.
 
-```json {% fileName="packages/some-project/project.json" highlightLines=[6] %}
-{
-  "name": "some-project",
-  ...
-  "targets": {
-    "build": {
-      "inputs": ["..."],
-      ...
-    }
-    ...
+```jsonc {% highlightLines=[6] %}
+"targets": {
+  "lint": {
+    "command": "eslint .",
+    "inputs": [
+      "default",
+      { "externalDependencies": ["eslint", "eslint-config-airbnb"] }
+    ]
   }
 }
 ```
 
-{% /tab %}
+Targets which use an Executor of Nx Plugins maintained by the Nx Team will have the correct set of external dependencies considered when calculating the hash.
 
-{% tab label="Project Level (package.json)" %}
+By explicitly providing an empty array, Nx ignores all changes to external dependencies.
+
+### Outputs of Dependent Tasks
+
+When a task depends on another task, it is possible that the outputs of the dependent task will affect the behavior of the task. Nx can consider the contents of files produced by dependent tasks matching a pattern when calculating the computation hash. Outputs of Dependent Tasks as inputs are defined as such:
+
+```jsonc
+"inputs": [
+  { "dependentTasksOutputFiles": "**/*.d.ts" },
+  { "dependentTasksOutputFiles": "**/*.d.ts", "transitive": true }
+]
+```
+
+The pattern will be matched with outputs of tasks which this task depends on. Setting `transitive` to `true` will also include outputs of all task dependencies of this task in the [task pipeline](/concepts/task-pipeline-configuration).
+
+### A subset of the root `tsconfig.json` or `tsconfig.base.json`
+
+When a root `tsconfig.json` or `tsconfig.base.json` is present, Nx will always consider parts of the file which apply to the project of a task being run.
+This includes the full `compilerOptions` section and particular path mappings in the `compilerOptions.paths` property.
+This allows Nx to to not invalidate every single task when a path mapping is added or removed from the root `tsconfig.json` file
 
-```json {% fileName="packages/some-project/package.json" highlightLines=[9] %}
+## Named Inputs
+
+Many tasks will utilize the same sets of inputs with minor differences. Nx allows you to define these sets in the `namedInputs` section of the `nx.json` file.
+
+These sets of inputs can then be referenced and reused in the `inputs` array of targets. You can think of these as variables which can be used in the `inputs` array.
+
+Named Inputs can be defined for the entire workspace in `nx.json` or for a specific project in `project.json`.
+
+Once defined for the workspace, any tasks for all projects in the workspace can use the named inputs in the `inputs` array.
```

Comment:
We should mention that Nx will always include a `"default": "{projectRoot}/**/*"` named input, even if its not specified. But it will be overridden if it is specified.